internal/shim-sgx/src/handler/gdb.rs

```diff
@@ -195,81 +170,35 @@ impl<'a> gdbstub::Connection for super::Handler<'a> {
     }
 
     fn write_all(&mut self, buf: &[u8]) -> Result<(), Self::Error> {
-        let bytes_len = buf.len();
-        let mut to_write = bytes_len;
-
-        loop {
-            let next = bytes_len.checked_sub(to_write).ok_or(libc::EFAULT)?;
-
-            let [written, _] = self.syscall(
-                Register::<usize>::from(buf[next..].as_ptr()),
-                Register::<usize>::from(to_write),
-                0usize.into(),
-                0usize.into(),
-                0usize.into(),
-                0usize.into(),
-                SYS_ENARX_GDB_WRITE as _,
-            )?;
-
-            // be careful with `written` as it is untrusted
-            to_write = to_write.checked_sub(written.into()).ok_or(libc::EIO)?;
-            if to_write == 0 {
-                break;
-            }
-        }
-
+        self.gdb_write_all(buf)?;
         Ok(())
     }
 
     fn peek(&mut self) -> Result<Option<u8>, Self::Error> {
-        let [val, _] = self.syscall(
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            SYS_ENARX_GDB_PEEK as _,
-        )?;
-
-        Ok(u8::try_from(usize::from(val)).ok())
+        self.gdb_peek()
     }
 
     fn flush(&mut self) -> Result<(), Self::Error> {
         Ok(())
     }
 
     fn on_session_start(&mut self) -> Result<(), Self::Error> {
-        let [_, _] = self.syscall(
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            0usize.into(),
-            SYS_ENARX_GDB_START as _,
-        )?;
-
-        Ok(())
+        self.gdb_on_session_start()
     }
 }
 
 #[derive(Debug)]
 pub(crate) struct GdbTarget {
     regs: X86_64CoreRegs,
-    shim_range: Range<*const u8>,
```

Comment:
Why this change?

Reply:
I don't want to do pointer arithmetic and always make `unsafe` safety comments, because I don't plan to de-reference the pointers here.